src/main/java/com/example/bookstore/product/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.example.bookstore.product.service;
+
+import com.example.bookstore.product.Repository.ProductRepository;
+import com.example.bookstore.product.entity.Product;
+import com.example.bookstore.product.model.ProductInput;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class ProductServiceImpl implements ProductService {
+
+    private final ProductRepository productRepository;
+
+    @Override
+    public void add(ProductInput parameter) {
+        Product product = Product.builder()
+                .productName(parameter.getProductName())
+                .productType(parameter.getProductType())
+                .price(parameter.getPrice())
+                .salePrice(parameter.getSalePrice())
+                .productDetail(parameter.getProductDetail())
+                .stock(parameter.getStock())
+                .regDate(LocalDateTime.now())
+                .build();
+
+        productRepository.save(product);
+    }
+
+    @Override
+    public List<Product> list() {
+
+        //Todo 일단 모든 데이터 가져오고 추후 필요에 따라 수정 or Elasticsearch
+
+        return productRepository.findAll();
+    }
+
+    @Override
+    public Product detail(String id) {
+
+        Optional<Product> optionalProduct = productRepository.findById(Long.parseLong(id));
+
+        return optionalProduct.orElse(null);
+    }
+
+    @Override
+    public void update(ProductInput parameter, String id) {
+        Optional<Product> optionalProduct = productRepository.findById(Long.parseLong(id));
+
+        if(optionalProduct.isEmpty()) {
+            throw new IllegalStateException();
+        }
```

Comment:
변수를 Optional<x>로 받는 형태를 지양해주세요
```suggestion
        Product product = productRepository.findById(Long.parseLong(id)).orElseThrow(() -> new IllegalStateException());
```